src/set_theory/ordinal/basic.lean

```diff
@@ -1180,25 +1155,30 @@ by simp only [lift.principal_seg_top, univ_id]
 
 /-! ### Minimum -/
 
+section argmin_min
+
+variables {ι : Sort*} {I : nonempty ι} {f : ι → ordinal}
+
+/-- The index of the minimal element of a nonempty family of ordinals. -/
+def argmin (I : nonempty ι) (f : ι → ordinal) : ι :=
+(lt_wf.min_mem (set.range f) (@set.range_nonempty _ _ I f)).some
+
 /-- The minimal element of a nonempty family of ordinals -/
-def min {ι} (I : nonempty ι) (f : ι → ordinal) : ordinal :=
-lt_wf.min (set.range f) (let ⟨i⟩ := I in ⟨_, set.mem_range_self i⟩)
+def min (I : nonempty ι) (f : ι → ordinal) : ordinal := f (argmin I f)
```

Comment:
I also don't think there's a need for this, since we already have `infi`. In fact, I already have #14707 for completely phasing this definition out.

Reply:
What do you think about defining `ord` and `well_ordering_rel` using `function.argmin`?

Reply:
Anyway, I'm going to close this PR in about a day or two, possibly moving the bits about `galois_connection` to a new PR. Feel free to cherry-pick the definitions of `well_ordering_rel` and `ord` or ignore this approach.